[Kah Leong] iP

kahleongq opened the pull request:
### Duke

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Duke frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,


1. download it from [here](https://nus-cs2103-ay2122s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [ ]  Managing tasks 
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Duke {
    public static void main(String[] args) throws IOException {
        new Duke("data/tasks.txt").run();
    }
}
```

Commit:
Add Gradle support

Commit:
Level-1 commit

Commit:
Level-2 commit

Commit:
Level-3 commit

Commit:
Level-4 commit

Commit:
text UI testing commit

Commit:
Level-5 commit

Commit:
minor changes to level-5

Commit:
Level-6 commit

robinrojh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,124 @@
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        String nextLine = br.readLine().strip().replaceAll(" +", " ");
+
+        ArrayList<Task> list = new ArrayList<Task>();
```
Using a longer variable name (e.g. taskList) would help in case you need to change the calls for this variable!

robinrojh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,124 @@
+            String firstWord = nextLine.split(" ")[0];
+            if (firstWord.equals("mark")) {
+                try {
+                    int indexToMark = Integer.parseInt(nextLine.substring(5)) - 1;
```
In cases like this, it could be cleaner to keep the variable name shorter as the scope of this variable is only in the try block.

robinrojh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,124 @@
+                }
+            } else if (firstWord.equals("unmark")) {
+                try {
+                    int indexToUnmark = Integer.parseInt(nextLine.substring(7)) - 1;
```
Same here! However, you can decide for yourself. If you feel like you will use this variable many times and in deeper blocks, you can keep it this way.

robinrojh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,124 @@
+            + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
```
Since the BuefferedReader br has a very large scope (accessible everywhere main method), you should most likely name this a bit longer so that it is easier to modify or search for in the future.

robinrojh reviewed (commented):
The only part that you seem to miss is the naming of variables-- storing variables with a larger scope with a longer name and storing variables with a smaller scope with a shorter name. Overall, you have met most of the coding standards.

simjunheng commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,124 @@
+                        System.out.println((i + 1) + "." + list.get(i));
+                    }
+                }
+            } else if (firstWord.equals("delete")) {
```
Perhaps it would be good if you can create more classes e.g. Parser with specific purposes. This way, you might be able to avoid deep nesting

simjunheng commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,28 @@
+public class Event extends Task {
```
Maybe it would be a good idea to include class description? This way, fellow developers can catch a glimpse of the class purpose

simjunheng commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,28 @@
+        this.at = at;
+    }
+    
+    @Override
```
Perhaps it would be a good idea to include method descriptions

simjunheng reviewed (commented):
The absence of java docs and code quality violations may be due to the incompletion of week 3 IP tasks. These problems should naturally be gone once week 3 tasks are completed. You can do it! 👍

kahleongq committed:
Update Duke to save list in hard disk

kahleongq committed:
Update tasks to handle dates

kahleongq committed:
Merge tag 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

kahleongq committed:
Remove redundant lines

KristopherPTaslim commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,198 @@
+                int tempDescLength = tempDescription.length();
+                String description = tempDescription.substring(0, tempDescLength - 2);
+                String tempTimeBy = line.split("by: ")[1];
+                int endIdx = tempTimeBy.lastIndexOf(")");
```
perhaps it would be better to write endIndex instead of the short form.

KristopherPTaslim commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,198 @@
+                int tempDescLength = tempDescription.length();
+                String description = tempDescription.substring(0, tempDescLength - 2);
+                String tempTimeBy = line.split("at: ")[1];
+                int endIdx = tempTimeBy.lastIndexOf(")");
```
perhaps it would be better to write endIndex instead of the short form.

KristopherPTaslim commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,35 @@
+
+    public Event(String description, LocalDate at, boolean isDone) {
+        super(description, isDone);
+        this.at = at;
```
maybe can choose different variable naming cause 'at' might be ambiguous

KristopherPTaslim commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,35 @@
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = LocalDate.parse(by);
```
perhaps can change the variable naming of by

KristopherPTaslim reviewed (commented):
Overall, a relatively well variable naming and java coding standard are followed well! Goodluck on the incoming tasks.

kahleongq committed:
Add more classes to utilise more OOP, and add DukeException class.

kahleongq committed:
Package all classes under duke package.

kahleongq committed:
Fix TaskList handling dates from storage.

kahleongq committed:
Add JavaDoc comments to all classes.

kahleongq committed:
Fix files to follow coding standard

kahleongq committed:
Add find feature to find tasks

kahleongq committed:
Merge branch 'branch-A-CodingStandard'

kahleongq committed:
Merge branch 'branch-Level-9'

kahleongq committed:
Update find to accept phrases, and add javadocs.

kahleongq committed:
Fix bugs in parser

kahleongq committed:
Update todo Javadoc

kahleongq committed:
Add JUnit tests for parser and tasks

kahleongq committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

kahleongq committed:
Add gradle support to be able to build and run Duke, and checkstyle

kahleongq committed:
Duke is a task manager that can only be used in the command line.

However, more people prefer the use of a GUI instead of CLI.

Let's implement a GUI through the use of JavaFX.

kahleongq committed:
Add assertions to codebase

kahleongq committed:
Merge pull request #2 from kahleongq/branch-A-Assertions

Add assertions to codebase

kahleongq committed:
Refactor code to reduce code duplication

kahleongq committed:
Merge pull request #3 from kahleongq/branch-A-CodeQuality

Refactor code to reduce code duplication

kahleongq committed:
Duke can only keep track of tasks.

Duke should be able to have more features to gain more users.

Let's add a new feature to allow users to keep track of certain places,
and record notes for these places.

kahleongq committed:
Finalize codebase

kahleongq committed:
Fix codebase to follow the coding standard

kahleongq committed:
Update to make GUI bigger

kahleongq committed:
Update GUI resolution to reduce amount of whitespace.

kahleongq committed:
Added user guide and screenshot of the UI.

kahleongq committed:
Update build.gradle file to have the correct main file

kahleongq committed:
Update build.gradle file to fix jar file from crashing
